janus_core/helpers/mlip_calculators.py

```diff
@@ -6,17 +6,72 @@
 - https://github.com/Quantum-Accelerators/quacc.git
 """
 
-from typing import get_args
+from pathlib import Path
+from typing import Any, Optional, Union, get_args
 
 from ase.calculators.calculator import Calculator
 import torch
 
-from janus_core.helpers.janus_types import Architectures, Devices
+from janus_core.helpers.janus_types import Architectures, Devices, PathLike
+
+
+def _set_model_path(
+    model_path: Optional[PathLike] = None,
+    kwargs: Optional[dict[str, Any]] = None,
+) -> Optional[Union[PathLike, torch.nn.Module]]:
+    """
+    Set `model_path`.
+
+    Parameters
+    ----------
+    model_path : Optional[PathLike]
+        Path to MLIP file.
+    kwargs : Optional[dict[str, Any]]
+        Dictionary of additional keyword arguments passed to the selected calculator.
+
+    Returns
+    -------
+    Optional[Union[PathLike, torch.nn.Module]]
+        Path to MLIP model file, loaded model, or None.
+    """
+    kwargs = kwargs if kwargs else {}
+
+    # kwargs that may be used for `model_path`` for different MLIPs
+    # Note: "model" for chgnet (but not mace_mp or mace_off) and "potential" may refer
+    # to loaded PyTorch models
+    model_path_kwargs = ("model", "model_paths", "potential", "path")
+
+    # Use model_path if specified, but check not also specified via kwargs
+    if model_path:
+        if any(kwarg in kwargs for kwarg in model_path_kwargs):
+            raise ValueError(
+                "`model_path` cannot be used in combination with 'model', "
+                "'model_paths', 'potential', or 'path'"
+            )
+    else:
+        # Check at most one suitable kwarg is specified
+        if sum(kwarg in kwargs for kwarg in model_path_kwargs) > 1:
+            raise ValueError(
+                "Only one of 'model', 'model_paths', 'potential', and 'path' can be "
+                "specified"
+            )
+
+        # Set model_path from kwargs if any are specified
+        for kwarg in model_path_kwargs:
+            if kwarg in kwargs:
+                model_path = kwargs.pop(kwarg, None)
```

Comment:
`kwargs` is a dictionary which means we can simplify this a lot using set operations.

`kwargs.keys()`  supports set operations, this means if we make `model_path_kwargs` a set too, we can check which kwargs are defined by:
```python
present = kwargs.keys() & {"model", "model_paths", "potential", "path"}
```
This leaves us with:
```suggestion
    present = kwargs.keys() & {"model", "model_paths", "potential", "path"}

    # Use model_path if specified, but check not also specified via kwargs
    if model_path and present:
        raise ValueError(
            "`model_path` cannot be used in combination with 'model', "
            "'model_paths', 'potential', or 'path'"
        )
    elif len(present) > 1:
        # Check at most one suitable kwarg is specified
        raise ValueError(
            "Only one of 'model', 'model_paths', 'potential', and 'path' can be "
            "specified"
        )
    elif present:
        # Set model_path from kwargs if any are specified
        model_path = kwargs[present[0]]
    else:
        return None
```
Alternatively you can check if `len(present) == 1` if you want to align, or set `model_path = None` if you want to have single-return.

Reply:
Thanks, that's useful! It did feel quite convoluted.

Reply:
I made a couple of changes to pop the `kwarg` when we set `model_path`, to avoid passing it twice, and removing the `return None`, to make sure `model_path` is converted into a `Path` as well.